cinema/serializers.py

```diff
@@ -68,13 +78,105 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+
+    def validate(self, attrs):
+        data = super(TicketSerializer, self).validate(attrs)
+        cinema_hall = attrs["movie_session"].cinema_hall
+
+        if not (1 <= attrs["seat"] <= cinema_hall.seats_in_row):
+            raise serializers.ValidationError(
+                f"seat number must be in range (1, {cinema_hall.seats_in_row})"
+                f", not {attrs['seat']}"
+            )
+        if not (1 <= attrs["row"] <= cinema_hall.rows):
+            raise serializers.ValidationError(
+                f"row number must be in range (1, {cinema_hall.rows})"
+                f", not {attrs['row']}"
+            )
+        return data
+
+    class Meta:
+        model = Ticket
+        fields = (
+            "id",
+            "row",
+            "seat",
+            "movie_session"
+        )
+
+
+class TicketListSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+
+class TicketDetailSerializer(TicketSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketDetailSerializer(
+        source="tickets",
+        many=True,
+        read_only=True
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = (
+            "id",
+            "show_time",
+            "movie",
+            "cinema_hall",
+            "taken_places"
+        )
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(
+        many=True,
+        read_only=False,
+        allow_empty=False,
+        allow_null=False
+    )
+
+    class Meta:
+        model = Order
+        fields = (
+            "id",
+            "tickets",
+        )
+
+    @transaction.atomic
+    def create(self, validated_data):
+        tickets_data = validated_data.pop("tickets")
+        order = Order.objects.create(**validated_data)
+        for ticket_data in tickets_data:
+            Ticket.objects.create(
+                order=order,
+                **ticket_data
+            )
```

Comment:
advice:
use python list generators and `bulk_create`, which would be more efficient than simple for loop